recipes/dd-opentracing-cpp/all/conanfile.py

```diff
@@ -0,0 +1,88 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class DatadogOpenTracingConan(ConanFile):
+    name = "dd-opentracing-cpp"
+    description = "Monitoring service for cloud-scale applications based on OpenTracing "
+    license = "Apache 2.0 License"
+    topics = ("instrumentration", "monitoring", "security", "tracing")
+    homepage = "https://github.com/DataDog/dd-opentracing-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/*.patch"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+
+        self.options["msgpack"].cpp_api = True
+
+    def requirements(self):
+        self.requires("opentracing-cpp/1.6.0")
+        self.requires("zlib/1.2.11")
+        self.requires("libcurl/7.73.0")
+        self.requires("msgpack/3.3.0")
+        self.requires("nlohmann_json/3.9.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
If  it ever (🤞) would require no patches.
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

Reply:
Perfect. I've added it with some other fixes